stdlib/filename.mli

```diff
@@ -183,9 +183,21 @@ val get_temp_dir_name : unit -> string
 (** The name of the temporary directory:
     Under Unix, the value of the [TMPDIR] environment variable, or "/tmp"
     if the variable is not set.
-    Under Windows, the value of the [TEMP] environment variable, or "."
-    if the variable is not set.
+
+    Under Windows, it uses [GetTempPath2] (since Windows 10 21H1), or
+    [GetTempPath], and returns:
+    - if the process runs as SYSTEM, the path "C:\\Windows\\SystemTemp\\"
+      (since Windows 10 21H1); or
+    - the path specified by the [TMP] environment variable; or
+    - the path specified by the [TEMP] environment variable; or
+    - the path specified by the [USERPROFILE] environment variable; or
+    - the Windows directory.
```

Comment:
Which directory is "the Windows directory"?

Reply:
OK, thanks.

Reply:
I'm not totally sure, but is this overly technical here - couldn't we just indicate that we're using `GetTempPath2` on Windows? The key thing to me is the fallback - we used to use `.` on failure (which is likely to work for an unprivileged user) and now we fallback to the Windows directory (which is not likely to work for an unprivileged user).

Reply:
This has been marked as resolved with neither an update nor a reply?

Reply:
> This has been marked as resolved with neither an update nor a reply?

Sorry, probably a mistake on my end.

> I'm not totally sure, but is this overly technical here - couldn't we just indicate that we're using `GetTempPath2` on Windows?

I was worried of being insufficiently technical here, but indicating that we're using `GetTempPath2` or `GetTempPath` is fine by me.

Reply:
I've applied your suggestion.

Reply:
(note that @nojb suggested the opposite earlier, that indicating the syscalls was a bit unusual for the ocaml documentation)

Reply:
Ah, the GH interface I fear was making that change difficult to perceive - sorry for oscillating you between two opinions!! I think it's better this way - Windows (as I've pontificated on various platforms) prefers syscalls to environment variables, so the fact we document the environment variables on Unix and the syscall on Windows seems consistent to me, @nojb? (this shouldn't spin any further - what @nojb says then goes!)

Reply:
Sounds good to me! Thanks.